### PR TITLE
Support PostgreSQL restore modes

### PR DESCRIPTION
**Pull request by zikalino**
##### SUMMARY

This PR implements new options for PostgreSQL server as requested by users.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
azure_rm_postgresqlserver

##### ANSIBLE VERSION
2.7

##### ADDITIONAL INFORMATION
NOTE: It's not possible to write integration test, as the backup may be not available for several hours after instance is created, so this options can be only tested manually at the moment.

### CODE REVIEW

File: lib/ansible/modules/cloud/azure/azure_rm_postgresqlserver.py [hunk 1]
```diff
@@ -71,7 +71,30 @@
     create_mode:
         description:
             - Create mode of SQL Server
-        default: Default
+        default: default
+        choices:
+            - default
+            - geo_restore
+            - point_in_time_restore
+    source_server_id:
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_postgresqlserver.py:79]: does it take dict of resource  group and name? if not, better to  support it.

Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_postgresqlserver.py:79]: not yet, I can do it to follow standard

**Comment by yuwzho** [comment]
Fixes #44806

**Comment by Fred-sun** [comment]
@zikalino Could you help to push this PR put forward? Thanks!

**Comment by Fred-sun** [comment]
@zikalino

**Comment by Fred-sun** [comment]
@zikalino Please help resolve the conflicting files. Thanks!

**Comment by Fred-sun** [comment]
@zikalino

**Comment by Fred-sun** [comment]
@zikalino could you help resolve the conflicting files? Thanks!

**Comment by Fred-sun** [comment]
@zikalino Please help resolve conflicting file and push to review. Thanks!

**Comment by Fred-sun** [comment]
@zikalino

**Comment by Fred-sun** [comment]
@zikalino

**Comment by Fred-sun** [comment]
@zikalino

**Comment by Fred-sun** [comment]
@zikalino Please help resolve conflicting file and push to review. Thanks!

**Comment by Fred-sun** [comment]
@zikalino

**Comment by Fred-sun** [comment]
@zikalino

**Comment by Fred-sun** [comment]
@yungezz @yuwzho review_need

**Comment by yungezz** [comment]
ci failed, pls fix ci first.

**Comment by Fred-sun** [comment]
@zikalino

**Comment by Andersson007** [comment]
@zikalino , hello, could you please complete this PR?
I would suggest help but i haven't work with azure at all and don't have a possibility to check it

needs_info

**Comment by Fred-sun** [comment]
@zikalino Please re-check and push PR to merged! Thanks!

**Comment by Fred-sun** [comment]
@zikalino Please help take a look this PR when you're available! ci fail, please give a fixes! Thanks a lot!

**Comment by Fred-sun** [comment]
@zikalino Please resolve the shippable error and push for review! Thank you very much!

**Pull request closed without merging**